CMakeLists.txt

```diff
@@ -272,7 +273,9 @@ endif()
 add_subdirectory(src)
 
 if(BUILD_SHARED_LIBS)
-  add_subdirectory(tools)
+  if(BUILD_TOOLS)
```

Comment:
`if (NOT ANDROID)`

Only add an option if user has both options.
We can handle Android build in the code instead letting library users know how to correctly configure the only option.

Reply:
what will happen if it's been build with termux, where `ANDROID` is defined.

Reply:
@fxliang yes. need some work to figure out the proper condition here.

Back on my argument why I prefer not to have the option.

Frist, unimportant, but immediately obvious to me, is the presence of the `if (BUILD_SAHRED_LIBS)` condition.
The current code is already making decision whether we build tools. Introducing an option leaves un in a conflicting mindset: it's not entirely up to the library user to decide if we are going to build tools; if `BUILD_SHARED_LIBS` is `OFF`, the cmake option is not respected. It's a breaking change if we require developers to additionally specify `BUILD_TOOLS:OFF` in their static librime build.

That introduces my second argument:
by providing an option to library users, librime developers are responsible for making sure both cases work. For now I believe none of us are interested in working on the BUILD_TOOLS=ON case on Android, so we don't want to take the responsibilty maintaining the option.

I don't think hard-coding a decision is a problem here. It's honest: the options you currently see are what we can provide as maintained use cases of the libarary product. So far it's able to meet every developers requirement; we don't forsee any use case in the other code path, like creating tools while building librime as static library - it might be possible, but as no one is known to want to use it, we don't bother maintaining that code path - maintaining a build variant by option is non-trivial work. Some day, when such requirement is real, it'll be a good time to test that code path, take responsibility in maintaining it, and introduce an option to unblock it for every developer.